### PR TITLE
Adding code for String question 

### PR DESCRIPTION
**Pull request by anilhariparamhans**
Calculate number of lines and number of char in last line

### CODE REVIEW

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+				countArray[i] = countArray[i] + 1;
+			}
+		}
+	}
+}
+
+STRINGPERMUTATION_API std::vector<std::string> getStringPermutations(std::string const & str)
+{
+	std::unordered_map<char, size_t> strMap;
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp]: If we want sorted output, we need to take map in place of unordered map.

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#include "StringPermutation.h"
+#include <unordered_map>
+#include <array>
+
+
+
+void getStringPermutations(std::vector<char> const &charArray, std::vector<size_t> &countArray, std::vector<std::string>& stringVec, size_t const strLenght, size_t level, std::string &result)
+{
+	if (level >= strLenght)
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp]: Please don't mind that I am pointing out this -> length spelling mistake in strLenght

File: LeetCode/vAnil/String/NumberOfLinesToWriteString/NumberOfLinesToWriteString.cpp [hunk 3]
```diff
@@ -1,10 +1,12 @@
+//https://leetcode.com/problems/number-of-lines-to-write-string/description/
 #include "NumberOfLinesToWriteString.h"
 
+
 NUMBEROFLINESTOWRITESTRING_API std::vector<size_t> numberOfLines(const std::vector<size_t>& widths, const std::string &s, size_t lineLen)
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/String/NumberOfLinesToWriteString/NumberOfLinesToWriteString.cpp:5]: NUMBEROFLINESTOWRITESTRING_API : how does compiler know if export line has to be defined here or import. Where are we defining if we have to export this function or import?

**Review by dimplenaresh1 — APPROVED** [review]
Looks good.

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
+#include "StringPermutation.h"
+#include <unordered_map>
+#include <array>
+
+
+
+void getStringPermutations(std::vector<char> const &charArray, std::vector<size_t> &countArray, std::vector<std::string>& stringVec, size_t const strLenght, size_t level, std::string &result)
+{
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp:8]: try...catch missing.

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+				getStringPermutations(charArray, countArray, stringVec, strLenght, level + 1, result);
+
+				countArray[i] = countArray[i] + 1;
+			}
+		}
+	}
+}
+
+STRINGPERMUTATION_API std::vector<std::string> getStringPermutations(std::string const & str)
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp]: std::string const & str is same as const std::string& str ???

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	std::vector<size_t> countArray;
+
+	for (auto val : strMap)
+	{
+		charArray.push_back(val.first);
+		countArray.push_back(val.second);
+	}
+
+	std::vector<std::string> stringVec;
+	std::string result = str;
```
Comment (dimplenaresh1) [inline LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp]: shouldn't result string be empty at first?

**Review by dimplenaresh1 — COMMENTED** [review]
Please review the comments.

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp (hunk 1 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp]: It is good to have lexicographical order sorted string.
I incorporated this review comment.

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp (hunk 2 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp]: This is good to point out this 👍 
I incorporated this review comment.

File: LeetCode/vAnil/String/NumberOfLinesToWriteString/NumberOfLinesToWriteString.cpp (hunk 3 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/String/NumberOfLinesToWriteString/NumberOfLinesToWriteString.cpp:5]: This is defined in header file:

#ifdef NUMBEROFLINESTOWRITESTRING_EXPORTS
#define NUMBEROFLINESTOWRITESTRING_API __declspec(dllexport)
#else
#define NUMBEROFLINESTOWRITESTRING_API __declspec(dllimport)
#endif

Comment (anilhariparamhans) [reply LeetCode/vAnil/String/NumberOfLinesToWriteString/NumberOfLinesToWriteString.cpp:5]: I incorporated this review comment.

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp (hunk 4 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp:8]: I incorporated this review comment.

File: LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp (hunk 6 above)
Comment (anilhariparamhans) [reply LeetCode/vAnil/String/StringPermutation/StringPermutation.cpp]: Excellent observation 👍 
Actually this was just a trick to make same length result string as input string to 
store the permutation of string. Now I am using different method.
I incorporated this review comment.

**Comment by anilhariparamhans** [comment]
@dimplenaresh1 Please review the code changes.

**Pull request merged**